Gnimuc/CImGui.jl
# 1.90.8 and backend shenanigans

I began the bindings update to 1.90.8 and ran into a bunch of things that needed fixing in our backends, so I decided to try using the official GLFW/OpenGL3 backends instead. And amazingly, they seem to work :octopus: This isn't ready to merge yet but I'm opening it for transparency. `demo/demo.jl` should work, but not `example/demo.jl` yet.

How to test it:
- Clone my `cimguipack-update` branch of Yggdrasil (https://github.com/JuliaPackaging/Yggdrasil/pull/8870)
- Build `CImGuiPack` locally with: `julia +1.7 build_tarballs.jl <platform> --deploy=local`. Change the platform to your desired one, for example (in a project with `BinaryBuilder` and `BinaryBuilderBase` installed):
  ```bash
  julia +1.7 --project=@bb build_tarballs.jl x86_64-linux-gnu --deploy=local # Linux
  julia +1.7 --project=@bb build_tarballs.jl x86_64-apple-darwin --deploy=local # OSX
  julia +1.7 --project=@bb build_tarballs.jl x86_64-w64-mingw32 --deploy=local # Windows
  ```

  If you get configure errors about `cxx_std_20`, pick a target that's in this CI list that matches your system: https://buildkite.com/julialang/yggdrasil/builds/11093. For example, on my system I had to use `x86_64-linux-gnu-cxx11-julia_version+1.11.0`.
- Clone this branch, `] dev ./LibCImGui.jl` and `] dev CImGuiPack`, and run `demo/demo.jl`. Docking and multiple viewports should work.
